apps/mobile-wallet/src/screens/Dashboard/useLoader.ts

```diff
@@ -36,13 +36,28 @@ export const useLoader = () => {
   const [showLoader, setShowLoader] = useState(addressesStatus === 'uninitialized')
 
   useEffect(() => {
-    if (progress === 1) {
-      setTimeout(() => setShowLoader(false), 500)
+    let timeoutID: NodeJS.Timeout
+
+    if (progress >= 1) {
+      timeoutID = setTimeout(() => setShowLoader(false), 300)
     }
+
+    return () => clearTimeout(timeoutID)
   }, [addressesStatus, progress])
 
   const updateProgress = () => setProgress((previousValue) => previousValue + 0.25)
 
+  useEffect(() => {
+    const intervalID = setInterval(() => {
+      if (progress < 1) {
+        // Fake progress ;)
+        setProgress((prev) => prev + 0.025)
+      }
+    }, 300)
+
+    return () => clearTimeout(intervalID)
+  })
```

Comment:
@mvaivre I'm a bit worried about changing the progress like this. If the API or the network is very slow to respond, the user will see a fake progress which might even end up reaching 100%... I would like to cap this to no more than 50%. WDYT?

Reply:
I will replace it with this, let me know if you disagree when you wake up :P

```t
useInterval(() => updateProgress(0.025), 300, progress >= 0.5)
```

Reply:
I had a better idea actually :)

```ts
  // Fake initial progress to show some movement for better UX ;)
  const stopFakeProgress = progress >= 0.9
  useInterval(() => setProgress((p) => p + 0.025 * (1 - p)), 300, stopFakeProgress)
```

Reply:
Done!

Reply:
NICE. Yes I was thinking about the need to cap this when I woke up :)